### ISSUE TITLE
`check-line-alignment` screws lists in tags

### ISSUE THREAD

**Issue by jwbth** [gajus/eslint-plugin-jsdoc#1541]
## Motivation

Lists may be used in `@param`, `@property`, or `@returns` ([random example](https://github.com/wikimedia/mediawiki/blob/26e679cfb2878f485ac29b8c523274a91ec0f702/resources/src/mediawiki.api/index.js#L211) in MediaWiki). Solutions like [Rewrap](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap), used by hundreds of thousands, automatically prettify those, making them look readable. They also support *nested* lists which should have correct indentation—otherwise their structure would be lost.

The `check-line-alignment` rule, on the other hand, is heavy-handed with lists: it either removes all indentation from the lines following the first line of the tag, or allows only fixed indentation with the `wrapIndent` parameter.

I suggest to implement any kind of solution that would prevent the plugin from screwing lists in tags:
- Specifically ignoring indentation of lines following list markup like `- `, `* `, `1. ` if the indentation is *greater* (not *less*) than the baseline
- Adding an option where you pass a regexp or function that would except lines from triggering the rule
- Fully supporting markdown lists (this may be asking for too much, but this would be a perfect solution where the plugin would do the work of Rewrap itself)

## Desired behavior

(From https://github.com/wikimedia/mediawiki/blob/26e679cfb2878f485ac29b8c523274a91ec0f702/resources/src/mediawiki.api/index.js#L211)

```js
/**
 * @return {mw.Api~AbortablePromise} A promise that settles when the API response is processed.
 *   Has an 'abort' method which can be used to abort the request.
 *   See {@link mw.Api~AbortablePromise} for an example.
 *
 *   - On success, resolves to `( result, jqXHR )` where `result` is the parsed API response.
 *   - On an API error, rejects with `( code, result, result, jqXHR )` where `code` is the
 *     [API error code](https://www.mediawiki.org/wiki/API:Errors_and_warnings), and `result`
 *     is as above. When there are multiple errors, the code from the first one will be used.
 *     If there is no error code, "unknown" is used.
 *   - On other types of errors, rejects with `( 'http', details )` where `details` is an object
 *     with three fields: `xhr` (the jqXHR object), `textStatus`, and `exception`.
 *     The meaning of the last two fields is as follows:
 *     - When the request is aborted (the abort method of the promise is called), textStatus
 *       and exception are both set to "abort".
 *     - On a network timeout, textStatus and exception are both set to "timeout".
 *     - On a network error, textStatus is "error" and exception is the empty string.
 *     - When the HTTP response code is anything other than 2xx or 304 (the API does not
 *       use such response codes but some intermediate layer might), textStatus is "error"
 *       and exception is the HTTP status text (the text following the status code in the
 *       first line of the server response). For HTTP/2, `exception` is always an empty string.
 *     - When the response is not valid JSON but the previous error conditions aren't met,
 *       textStatus is "parsererror" and exception is the exception object thrown by
 *       {@link JSON.parse}.
 */
```

## Current behavior

Note that sublists are killed.

```js
/**
 * @return {mw.Api~AbortablePromise} A promise that settles when the API response is processed.
 *   Has an 'abort' method which can be used to abort the request.
 *   See {@link mw.Api~AbortablePromise} for an example.
 *
 *   - On success, resolves to `( result, jqXHR )` where `result` is the parsed API response.
 *   - On an API error, rejects with `( code, result, result, jqXHR )` where `code` is the
 *   [API error code](https://www.mediawiki.org/wiki/API:Errors_and_warnings), and `result`
 *   is as above. When there are multiple errors, the code from the first one will be used.
 *   If there is no error code, "unknown" is used.
 *   - On other types of errors, rejects with `( 'http', details )` where `details` is an object
 *   with three fields: `xhr` (the jqXHR object), `textStatus`, and `exception`.
 *   The meaning of the last two fields is as follows:
 *   - When the request is aborted (the abort method of the promise is called), textStatus
 *   and exception are both set to "abort".
 *   - On a network timeout, textStatus and exception are both set to "timeout".
 *   - On a network error, textStatus is "error" and exception is the empty string.
 *   - When the HTTP response code is anything other than 2xx or 304 (the API does not
 *   use such response codes but some intermediate layer might), textStatus is "error"
 *   and exception is the HTTP status text (the text following the status code in the
 *   first line of the server response). For HTTP/2, `exception` is always an empty string.
 *   - When the response is not valid JSON but the previous error conditions aren't met,
 *   textStatus is "parsererror" and exception is the exception object thrown by
 *   {@link JSON.parse}.
 */
```

**Comment by brettz9** [comment]
@jwbth Do you want to take a look at #1601 to see if Co-pilot's fix is on the right track?

**Comment by jwbth** [comment]
@brettz9 Based purely on my reading of the changed files (without testing or having a close knowledge of the codebase), it is on the right track. It took a conservative approach, just making sure the nested lists aren't screwed, without trying to analyze the indentation further than that, let alone transforming it.
